### PR TITLE
[`Jetmoe`] Fix RoPE

### PR DESCRIPTION
**Pull request by vasqu**
Seems like the integration tests on our CI also died for a while.

Fixes the rope dimension for jetmoe by setting a respective attribute mapping - the normal calculation is not valid, ie `hidden_dim / num_attn_heads`. For reference, why this is valid see https://github.com/huggingface/transformers/blob/895b3ebe418ebcf6a37fc838ff0effdb69d98386/src/transformers/models/jetmoe/modeling_jetmoe.py#L494
There could be better solutions but not sure if it's worth the effort. 
Fixes #40817

cc @gante @ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Was a test missing or just not being fetched ? (if you can check if we need a patch for this would be nice!)

**Comment by vasqu** [comment]
run-slow: jetmoe

**Comment by vasqu** [comment]
Looks like our integration tests were just failing which would've indicated this - not sure how long it has been like this tbh

The set values in the common tests were good enough so that this wasnt discovered. Checking in if the slow tests are run now; will likely add a for patch label here

**Comment by vasqu** [comment]
run-slow: jetmoe

File: src/transformers/models/jetmoe/modeling_jetmoe.py [hunk 1]
```diff
@@ -932,15 +932,6 @@ def forward(
...
 
-        if attention_mask is not None and self._attn_implementation == "flash_attention_2" and use_cache:
-            batch_size = inputs_embeds.shape[0]
-            is_padding_right = attention_mask[:, -1].sum().item() != batch_size
-            if is_padding_right:
-                raise ValueError(
-                    "You are attempting to perform batched generation with padding_side='right'"
-                    " this may lead to unexpected behaviour for Flash Attention version of JetMoe. Make sure to "
-                    " call `tokenizer.padding_side  = 'left'` before tokenizing the input. "
-                )
```
Comment (vasqu) [inline src/transformers/models/jetmoe/modeling_jetmoe.py:943]: We already have this https://github.com/huggingface/transformers/blob/dfae7dd98ddbef9ce1df4a6eecab35b0eff4dac6/src/transformers/generation/utils.py#L2414-L2427

Doesn't make sense to keep (fails the test occasionally as we can't guarantee it with randomly initialized masks)

**Comment by vasqu** [comment]
run-slow: jetmoe

**Comment by vasqu** [comment]
All tests pass, no more killed processes! Merging

**Comment by gante** [comment]
Thank you for fixing 🙏
